### DISCUSSION TITLE
Want to know the mesh data structure

### DISCUSSION THREAD

**Discussion by Ernest861** [simnibs/simnibs#439] [2. General]
Dear SimNIBS group,

Thanks for your work, it makes my study more enjoyable.
I have 3 questionnaires:
1. When I used subj2mni or mesh2nii, there would be skull bones. How to achieve this will affect the results? 
    I used ANT to remove the skull bones before using charm or charm_tms to build individual meshes. 

2. I tried to build pos.matsimnibs in matlab (because my pseudo stimulation was to reverse the coil), but failed. 
      How to generate the corresponding individual positions for each subject in batches way? 

3. I am curious about the three-column data of gray_matter.elemetn_data{3,1}.tetdata, because I found that only the first column seems to be needed to calculate the average value, so what are the values ​​of the remaining two columns? 
`field = gray_matter.element_data{field_idx}.tetdata;
avg_field_roi = sum(field(roi) .* elm_vols(roi))/sum(elm_vols(roi));`

Best regards,
Hui

**Comment by bjornsig** [comment]
1. Can you please explain why you need your mesh without the skull?
2. Please see here. This is how matsimnibs is calculated in simnibs from a position and a reference coordinate. https://github.com/simnibs/simnibs/blob/d045e8ccda6b349cbec639563af32dac69e95909/simnibs/mesh_tools/mesh_io.py#L1866
3. Can you point to the example code where is this code `gray_matter.elemetn_data{3,1}.tetdata` is used?
